### ISSUE TITLE
harfbuzz failed to build on 10.6.8 

### ISSUE THREAD

**Issue by STudio26** [Homebrew/homebrew#43056]
The error seems to be in source code.
Does the error below means that the error is in /usr/include/dispatch/queue.h ?

  GISCAN   HarfBuzz-0.0.gir
/usr/include/dispatch/queue.h:67: syntax error, unexpected ';' in 'typedef struct dispatch_queue_s *dispatch_queue_t;;' at ';'

/usr/include/dispatch/queue.h seems quite standard in OS X, right?
Here is the line 67: DISPATCH_DECL(dispatch_queue);
It is like the compiler thinks that there are two comma...

Gist is available here: https://gist.github.com/anonymous/1b9f6f667b20bdebae3f

**Comment by tschoonj** [comment]
This is interesting:

````
dyld: Library not loaded: @loader_path//usr/local/opt/libpng/lib/libpng16.16.dylib
  Referenced from: /usr/local/opt/freetype/lib/libfreetype.6.dylib
  Reason: image not found
````

Looks like libpng and/or freetype are not installed (or not properly at least).

Did you run `brew upgrade` before installing harfbuzz?
This message may be fixed by reinstalling freetype.
Apart from this, the main problem appears to be with gobject-introspection. You may want to remove it from the dependency list and add `--enable-introspection=no` the configure argument list in the formula.

**Comment by STudio26** [comment]
I ran `brew upgrade` before and got a lot of updates. Most of them were ok, I got to download manually the packages for some of them (curl did not recognise the redirection). Before that I made a XCode upgrade to the latest available for OS 10.6.8.
libpng and feetype are installed according to `brew info libpng` and `brew info freetype`. I will first reinstall freetype and then have a look at gobject-introspection if it does not work.
